### PR TITLE
Issue #1513 Document how to write child device config management agent

### PR DESCRIPTION
**Pull request by albinsuresh**
## Proposed changes

Document how to write child device config management agent
## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: docs/src/howto-guides/child_device_config_managament.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+```json
+{
+    "status": "successful",
+    "type": "{config-type}",
+    "path": "/child/local/fs/path" 
+}
+```
+
+If there are any failures while reading or uploading the requested conifg file,
```
Comment (didier-wenzek) [inline docs/src/howto-guides/child_device_config_managament.md:119]:
```suggestion
If there are any failures while reading or uploading the requested config file,
```

File: docs/src/howto-guides/child_device_config_managament.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    "type": "{config-type}",
+    "path": "/child/local/fs/path" 
+}
+```
+
+# Handle config update requests from thin-edge
+
+Handling config update requests from thin-edge is a 5-step process:
+
+1. Subsribe to, and receive conifg update requests via MQTT
```
Comment (didier-wenzek) [inline docs/src/howto-guides/child_device_config_managament.md:134]:
```suggestion
1. Subscribe to, and receive config update requests via MQTT
```

File: docs/src/howto-guides/child_device_config_managament.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+{
+    "status": "failed",
+    "type": "{config-type}",
+    "path": "/child/local/fs/path" 
+}
+```
+
+## Reference Implementation
+
+[Here](https://github.com/thin-edge/thin-edge.io_spikes/blob/main/file-transfer-service/child_device.py) is a reference child device agent implementaion written in Python to demonstrate the above mentioned protocol.
```
Comment (didier-wenzek) [inline docs/src/howto-guides/child_device_config_managament.md:203]: This example must move to the [example repository](https://github.com/thin-edge/thin-edge.io_examples),
the spike repository being private.

File: docs/src/howto-guides/child_device_config_managament.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+
+```json
+{
+    "status": "failed",
+    "type": "{config-type}",
+    "path": "/child/local/fs/path" 
+}
+```
+
+## Reference Implementation
```
Comment (didier-wenzek) [inline docs/src/howto-guides/child_device_config_managament.md:201]: I would also add a link to the reference documentation.

```suggestion
## References

 * [documentation](...)
 * [implementation example](...)
```

**Review by didier-wenzek — COMMENTED** [review]
Clear writing. Might be improved with some more context in the introduction, though.

* The spell checker report typos.
* `mdbook` checker reports markdown issues.
* The `SUMMARY.md` must be updated.

File: docs/src/howto-guides/child_device_config_managament.md (hunk 3 above)
Comment (albinsuresh) [reply docs/src/howto-guides/child_device_config_managament.md:203]: Done in this PR: https://github.com/thin-edge/thin-edge.io_examples/pull/36

**Review by didier-wenzek — APPROVED** [review]
Approved. A really nice documentation, clear and pleasant reading.

**Pull request merged**